core/iwasm/common/wasm_c_api.c

```diff
@@ -258,6 +258,12 @@ WASM_DEFINE_VEC_OWN(module, wasm_module_delete_internal)
 WASM_DEFINE_VEC_OWN(store, wasm_store_delete)
 WASM_DEFINE_VEC_OWN(valtype, wasm_valtype_delete)
 
+#ifndef NDEBUG
+#define WASM_C_DUMP_PROC_MEM() LOG_PROC_MEM()
+#else
+#define WASM_C_DUMP_PROC_MEM()
```

Comment:
Had better `#define WASM_C_DUMP_PROC_MEM() (void)0`, or when we call it with `WASM_C_DUMP_PROC_MEM();`, there will be an unused `;`, which might cause compilation warning.